packages/ckeditor5-table/tests/tablelayout/commands/tabletypecommand.js

```diff
@@ -64,6 +64,163 @@ describe( 'TableTypeCommand', () => {
+				setModelData( model, '<paragraph>f[o]o</paragraph>' );
+				expect( command.value ).to.be.null;
+			} );
+
+			it( 'should equal table type attribute value if selection is inside table', () => {
+				setModelData(
+					model,
+					'<table tableType="layout">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>f[o]o</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>'
+				);
+				expect( command.value ).to.equal( 'layout' );
+
+				setModelData(
+					model,
+					'<table tableType="content">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>f[o]o</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>'
+				);
+				expect( command.value ).to.equal( 'content' );
+			} );
+
+			it( 'should equal table type attribute value if selection is over table', () => {
+				setModelData(
+					model,
+					'[<table tableType="layout">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>foo</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>]'
+				);
+				expect( command.value ).to.equal( 'layout' );
+
+				setModelData(
+					model,
+					'[<table tableType="content">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>foo</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>]'
+				);
+				expect( command.value ).to.equal( 'content' );
+			} );
+
+			it( 'should equal table type attribute value if selection is inside a table caption', () => {
+				setModelData(
+					model,
+					'<table tableType="layout">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>[]foo</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>'
+				);
+
+				expect( command.value ).to.equal( 'layout' );
+
+				setModelData(
+					model,
+					'<table tableType="content">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>foo</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+						'<caption>bar[]baz</caption>' +
+					'</table>'
+				);
+
+				expect( command.value ).to.equal( 'content' );
+			} );
+
+			it( 'should equal table type attribute value if multiple table cells are selected', () => {
+				setModelData(
+					model,
+					'<table tableType="layout">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>[foo</paragraph>' +
+							'</tableCell>' +
+							'<tableCell>' +
+								'<paragraph>bar]</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>'
+				);
+
+				expect( command.value ).to.equal( 'layout' );
+
+				setModelData(
+					model,
+					'<table tableType="content">' +
+						'<tableRow>' +
+							'<tableCell>' +
+								'<paragraph>[foo</paragraph>' +
+							'</tableCell>' +
+							'<tableCell>' +
+								'<paragraph>bar]</paragraph>' +
+							'</tableCell>' +
+						'</tableRow>' +
+					'</table>'
+				);
+				expect( command.value ).to.equal( 'content' );
```

Comment:
```suggestion
				setModelData(
					model,
					'<table tableType="layout">' +
						'<tableRow>' +
							'[<tableCell>' +
								'<paragraph>foo</paragraph>' +
							'</tableCell>]' +
							'[<tableCell>' +
								'<paragraph>bar</paragraph>' +
							'</tableCell>]' +
						'</tableRow>' +
					'</table>'
				);

				expect( command.value ).to.equal( 'layout' );

				setModelData(
					model,
					'<table tableType="content">' +
						'<tableRow>' +
							'[<tableCell>' +
								'<paragraph>foo</paragraph>' +
							'</tableCell>]' +
							'[<tableCell>' +
								'<paragraph>bar</paragraph>' +
							'</tableCell>]' +
						'</tableRow>' +
					'</table>'
				);
				expect( command.value ).to.equal( 'content' );
```
The original selection proposed in this test was illegal as it crossed the limit element boundaries. It was post-fixed by the engine and could lead to false positive results.